### PR TITLE
DF-643:Archive details resolve links

### PR DESCRIPTION
**Pull request by TNA-Allan**
Ticket URL: https://national-archives.atlassian.net/browse/DF-643

## About these changes

1. resolves links from summary
2. removes links from fax field
3. resolves selected year dropdown link for "no year" for Accessions

## How to check these changes

Where possible, provide guidance to help your reviewer

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: templates/includes/records/archive-collections.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
         <h2>Collections information</h2>
             {% for collection in record.archive_collections.collection_info_list %}
                 {% if collection.count > 0 %}
-                    <h3>{{ collection.long_display_name }}({{ collection.count }})</h3>                    
+                    <h3  id="{{collection.name}}">{{ collection.long_display_name }}({{ collection.name }})</h3>
```
Comment (ababic) [inline templates/includes/records/archive-collections.html:6]:
```suggestion
                    <h3  id="{{collection.name}}">{{ collection.long_display_name }} ({{ collection.name }})</h3>                    
```

File: templates/includes/records/archive-description.html [hunk 2]
```diff
@@ -59,7 +59,7 @@ <h2>Summary of collections</h2>
                     <ul>
                         {% for collection in record.archive_collections.collection_info_list %}
                             {% if collection.count > 0 %}
-                                <li><a href="">{{ collection.display_name }}({{ collection.count }})</a></li>
+                                <li><a href="#{{collection.name}}">{{ collection.display_name }}({{ collection.count }})</a></li>
```
Comment (ababic) [inline templates/includes/records/archive-description.html:62]:
```suggestion
                                <li><a href="#heading-{{collection.name}}">{{ collection.display_name }}({{ collection.count }})</a></li>
```

File: templates/includes/records/archive-additional-resources.html [hunk 3]
```diff
@@ -6,7 +6,6 @@ <h2>Accessions</h2>
         <form action="" id="archive" method="post">
             <label class="archive-additional-resources__label" for="accessionYear">Select a year</label>
             <select class="archive-additional-resources__select" name="accessionYear" onchange="updateAction()" id="accessionYear">
-                <option value="">--Please choose an option--</option>
```
Comment (ababic) [inline templates/includes/records/archive-additional-resources.html:9]: What are the UI implications here? Won't the first year be selected by default? Is that necessarily desirable?

Comment (ababic) [reply templates/includes/records/archive-additional-resources.html:9]: Adding a `required` attribute to the `select` element should prevent form submission if the default is selected

File: templates/includes/records/archive-collections.html [hunk 4]
```diff
@@ -3,7 +3,7 @@
         <h2>Collections information</h2>
             {% for collection in record.archive_collections.collection_info_list %}
                 {% if collection.count > 0 %}
-                    <h3>{{ collection.long_display_name }}({{ collection.count }})</h3>                    
+                    <h3  id="heading-{{collection.name}}">{{ collection.long_display_name }}({{ collection.count }})</h3>
```
Comment (ababic) [inline templates/includes/records/archive-collections.html:6]: @TNA-Allan Any chance of getting a space after the `long_display_name` (as suggested previously)?

File: templates/includes/records/archive-description.html [hunk 5]
```diff
@@ -59,7 +59,7 @@ <h2>Summary of collections</h2>
                     <ul>
                         {% for collection in record.archive_collections.collection_info_list %}
                             {% if collection.count > 0 %}
-                                <li><a href="">{{ collection.display_name }}({{ collection.count }})</a></li>
+                                <li><a href="#heading-{{collection.name}}">{{ collection.display_name }}({{ collection.count }})</a></li>
```
Comment (ababic) [inline templates/includes/records/archive-description.html:62]: @TNA-Allan Any chance of getting a space after the `display_name` here (as suggested previously)?

File: templates/includes/records/archive-collections.html (hunk 4 above)
Comment (TNA-Allan) [reply templates/includes/records/archive-collections.html]: Sorry I missed it out. Done now.

File: templates/includes/records/archive-additional-resources.html (hunk 3 above)
Comment (TNA-Allan) [reply templates/includes/records/archive-additional-resources.html:9]: No JS 👍   Thanks @ababic

Comment (TNA-Allan) [reply templates/includes/records/archive-additional-resources.html:9]: The update is good for now. Thanks.

File: templates/includes/records/archive-description.html [hunk 6]
```diff
@@ -59,7 +59,7 @@ <h2>Summary of collections</h2>
                     <ul>
                         {% for collection in record.archive_collections.collection_info_list %}
                             {% if collection.count > 0 %}
-                                <li><a href="">{{ collection.display_name }}({{ collection.count }})</a></li>
+                                <li><a href="#heading-{{collection.name}}">{{ collection.display_name }} ({{ collection.count }})</a></li>
```
Comment (ababic) [inline templates/includes/records/archive-description.html:62]: Sorry @TNA-Allan, one last thing... is there a reason why we use `display_name` here, but `long_display_name` above?

Comment (TNA-Allan) [reply templates/includes/records/archive-description.html:62]: @andy 
One of the [collections ](https://github.com/nationalarchives/ds-wagtail/blob/3a374dcd2fa1033718ccc2c41e0f0f0b5b024dc7/etna/ciim/constants.py#L714-L755)(NRA records) have a `long_display_name` to show in the Collections Information (below) area and `display_name` (above) in the summary area

Comment (TNA-Allan) [reply templates/includes/records/archive-description.html:62]: For other collections, both values `long_display_name, display_name` are same.

**Review by ababic — APPROVED** [review]
Thanks for confirming on those last bits, Allan. LGTM!

**Pull request merged**